assets_src/assets/js/ContentElements/QueryBuilder.js

```diff
@@ -263,18 +262,32 @@
       return this._value;
     };
 
+    QueryBuilderRule.prototype.sanitizeValue = function (value)
+    {
+      // process using the valueObject
+      var valueObject = getInput.call(this);
+      if(valueObject && valueObject.sanitize)
+      {
+        value = valueObject.sanitize(value);
+      }
+      return value;
+    };
+
     QueryBuilderRule.prototype.setValue = function (value)
     {
-      this._setValue(value);
-      this.render();
+      this._setValue(this.sanitizeValue(value));
       $(this._queryBuilder._ele).trigger(
         'change.querybuilder', [this._queryBuilder.rules()]
       );
```

Comment:
This is more of a question than a critique;
Is there any need to trigger this event when it is also triggered in _setValue?